installer - support for Ubuntu 22.04 missing in PDF generator

- [x] unify pipelining and install with "sudo"
- [x] PDF Gen error
```
  Failed GeneratePdf [30 ms]
      Error Message:
       System.AggregateException : One or more errors occurred. (Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.)
```

- [x] fix error with slapd; error produced with

  - desktop version (ansible 2.10.8)
  - tim@fworch-comp:~/firewall-orchestrator-tpur$ slapd -V

       @(#) $OpenLDAP: slapd 2.5.11+dfsg-1~exp1ubuntu3 (Feb 16 2022 17:15:26) $
```
TASK [openldap-server : Initialize ldap config] **************************************************************************************************
fatal: [fworch-srv]: FAILED! => changed=true 
  cmd:
  - slapadd
  - -F
  - /etc/ldap/slapd.d
  - -n
  - '0'
  - -l
  - /etc/ldap/slapd.d/config.ldif
  delta: '0:00:00.008946'
  end: '2022-04-22 10:10:33.691683'
  msg: non-zero return code
  rc: 1
  start: '2022-04-22 10:10:33.682737'
  stderr: |-
    lt_dlopenext failed: (back_bdb.la) file not found
    slapadd: could not add entry dn="cn=module{0},cn=config" (line=11): <olcModuleLoad> handler exited with 1
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

```

might need to add config /etc/ldap/ldap.conf

    # Load dynamic backend modules:
    # modulepath xxx
    moduleload back_bdb.la

TASK [test : csharp tests] ********************************************************
fatal: [fworch-srv]: FAILED! => changed=true 
  cmd:
  - dotnet
  - test
  delta: '0:00:19.083354'
  end: '2022-04-25 13:23:14.004856'
  msg: non-zero return code
  rc: 1
  start: '2022-04-25 13:22:54.921502'
  stderr: ''
  stderr_lines: <omitted>
  stdout: |2-
      Determining projects to restore...
      Restored /usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj (in 9.45 sec).
      8 of 9 projects are up-to-date for restore.
      FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Debug/net6.0/FWO.Logging.dll
      FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Debug/net6.0/FWO.Config.File.dll
      FWO.Middleware -> /usr/local/fworch/lib/files/FWO.Middleware/bin/Debug/net6.0/FWO.Middleware.dll
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(31,21): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/APIConnection.cs(105,25): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
      FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Debug/net6.0/FWO.Api.Client.dll
      FWO.Middleware.Client -> /usr/local/fworch/lib/files/FWO.Middleware.Client/bin/Debug/net6.0/FWO.Middleware.Client.dll
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(43,16): warning CS8618: Non-nullable property 'User' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(84,34): warning CS8625: Cannot convert null literal to non-nullable reference type. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(30,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(32,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
      FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Debug/net6.0/FWO.Config.Api.dll
    /usr/local/fworch/lib/files/FWO.Report.Filter/Ast/AstNodeFilterDateTimeRange.cs(11,23): warning CS8618: Non-nullable field 'semanticValue' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Report.Filter/FWO.Report.Filter.csproj]
      FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Debug/net6.0/FWO.Report.Filter.dll
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(204,29): warning CS8618: Non-nullable property 'DomainList' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(63,57): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(96,67): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(142,82): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(143,25): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(97,38): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(117,47): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
      FWO.DeviceAutoDiscovery -> /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/bin/Debug/net6.0/FWO.DeviceAutoDiscovery.dll
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(19,16): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(51,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(17,23): warning CS0169: The field 'ApiTest.apiConnection' is never used [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
      FWO.Test -> /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll
    Test run for /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (.NETCoreApp,Version=v6.0)
    Microsoft (R) Test Execution Command Line Tool Version 17.1.0
    Copyright (c) Microsoft Corporation.  All rights reserved.
  
    Starting test execution, please wait...
    A total of 1 test files matched the specified pattern.
    2022-04-25T13:23:13+00:00 Info - Test Log (HtmlToPdfTest.cs in line 31), starting PDF generation
  
      Failed GeneratePdf [30 ms]
      Error Message:
       System.AggregateException : One or more errors occurred. (Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.)
      ----> System.NotSupportedException : Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.
      Stack Trace:
         at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
       at System.Threading.Tasks.Task.Wait()
       at WkHtmlToPdfDotNet.SynchronizedConverter.Invoke[TResult](Func`1 delegate)
       at WkHtmlToPdfDotNet.SynchronizedConverter.Convert(IDocument document)
       at FWO.Test.HtmlToPdf.HtmlToPdfTest.GeneratePdf() in /usr/local/fworch/test/csharp/FWO.Test/HtmlToPdfTest.cs:line 57
    --NotSupportedException
       at WkHtmlToPdfDotNet.ModuleFactory.GetModule()
       at WkHtmlToPdfDotNet.PdfTools.Load()
       at WkHtmlToPdfDotNet.BasicConverter.Convert(IDocument document)
       at WkHtmlToPdfDotNet.SynchronizedConverter.<>n__0(IDocument document)
       at WkHtmlToPdfDotNet.SynchronizedConverter.<>c__DisplayClass4_0.<Convert>b__0()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
    --- End of stack trace from previous location ---
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      Standard Output Messages:
     2022-04-25T13:23:13+00:00 Info - Test Log (HtmlToPdfTest.cs in line 31), starting PDF generation
  
  
  
    Failed!  - Failed:     1, Passed:    10, Skipped:     0, Total:    11, Duration: 169 ms - /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (net6.0)
  stdout_lines: <omitted>

`TASK [test : csharp tests] ********************************************************
fatal: [fworch-srv]: FAILED! => changed=true 
  cmd:
  - dotnet
  - test
  delta: '0:00:19.083354'
  end: '2022-04-25 13:23:14.004856'
  msg: non-zero return code
  rc: 1
  start: '2022-04-25 13:22:54.921502'
  stderr: ''
  stderr_lines: <omitted>
  stdout: |2-
      Determining projects to restore...
      Restored /usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj (in 9.45 sec).
      8 of 9 projects are up-to-date for restore.
      FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Debug/net6.0/FWO.Logging.dll
      FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Debug/net6.0/FWO.Config.File.dll
      FWO.Middleware -> /usr/local/fworch/lib/files/FWO.Middleware/bin/Debug/net6.0/FWO.Middleware.dll
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(31,21): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/APIConnection.cs(105,25): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
      FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Debug/net6.0/FWO.Api.Client.dll
      FWO.Middleware.Client -> /usr/local/fworch/lib/files/FWO.Middleware.Client/bin/Debug/net6.0/FWO.Middleware.Client.dll
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(43,16): warning CS8618: Non-nullable property 'User' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(84,34): warning CS8625: Cannot convert null literal to non-nullable reference type. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(30,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(32,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
      FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Debug/net6.0/FWO.Config.Api.dll
    /usr/local/fworch/lib/files/FWO.Report.Filter/Ast/AstNodeFilterDateTimeRange.cs(11,23): warning CS8618: Non-nullable field 'semanticValue' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Report.Filter/FWO.Report.Filter.csproj]
      FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Debug/net6.0/FWO.Report.Filter.dll
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(204,29): warning CS8618: Non-nullable property 'DomainList' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(63,57): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(96,67): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(142,82): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/AutoDiscoveryFortiManager.cs(143,25): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(97,38): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/CheckPointAPI.cs(117,47): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
      FWO.DeviceAutoDiscovery -> /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/bin/Debug/net6.0/FWO.DeviceAutoDiscovery.dll
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(19,16): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(51,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(17,23): warning CS0169: The field 'ApiTest.apiConnection' is never used [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
      FWO.Test -> /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll
    Test run for /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (.NETCoreApp,Version=v6.0)
    Microsoft (R) Test Execution Command Line Tool Version 17.1.0
    Copyright (c) Microsoft Corporation.  All rights reserved.
  
    Starting test execution, please wait...
    A total of 1 test files matched the specified pattern.
    2022-04-25T13:23:13+00:00 Info - Test Log (HtmlToPdfTest.cs in line 31), starting PDF generation
  
      Failed GeneratePdf [30 ms]
      Error Message:
       System.AggregateException : One or more errors occurred. (Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.)
      ----> System.NotSupportedException : Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.
      Stack Trace:
         at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
       at System.Threading.Tasks.Task.Wait()
       at WkHtmlToPdfDotNet.SynchronizedConverter.Invoke[TResult](Func`1 delegate)
       at WkHtmlToPdfDotNet.SynchronizedConverter.Convert(IDocument document)
       at FWO.Test.HtmlToPdf.HtmlToPdfTest.GeneratePdf() in /usr/local/fworch/test/csharp/FWO.Test/HtmlToPdfTest.cs:line 57
    --NotSupportedException
       at WkHtmlToPdfDotNet.ModuleFactory.GetModule()
       at WkHtmlToPdfDotNet.PdfTools.Load()
       at WkHtmlToPdfDotNet.BasicConverter.Convert(IDocument document)
       at WkHtmlToPdfDotNet.SynchronizedConverter.<>n__0(IDocument document)
       at WkHtmlToPdfDotNet.SynchronizedConverter.<>c__DisplayClass4_0.<Convert>b__0()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
    --- End of stack trace from previous location ---
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      Standard Output Messages:
     2022-04-25T13:23:13+00:00 Info - Test Log (HtmlToPdfTest.cs in line 31), starting PDF generation
  
  
  
    Failed!  - Failed:     1, Passed:    10, Skipped:     0, Total:    11, Duration: 169 ms - /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (net6.0)
  stdout_lines: <omitted>`

--- a/documentation/developer-docs/database/postgresql-debugging-using-pgadmin.md
+++ b/documentation/developer-docs/database/postgresql-debugging-using-pgadmin.md
@@ -22,10 +22,10 @@ the following was tested with Ubuntu 20.04
           sudo systemctl restart postgresql
 
 - on the clinet side:
-  - add pgadmin repo
+  - add pgadmin repo (here Ubuntu 22.04 - jammy)
 
           sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 8881B2A8210976F2
-          sudo sh -c "echo 'deb https://ftp.postgresql.org/pub/pgadmin/pgadmin4/apt/focal pgadmin4 main' > /etc/apt/sources.list.d/pgadmin4.list"
+          sudo sh -c "echo 'deb https://ftp.postgresql.org/pub/pgadmin/pgadmin4/apt/jammy pgadmin4 main' > /etc/apt/sources.list.d/pgadmin4.list"
           sudo apt update
 
   - install pgadmin

--- a/documentation/installer/install-for-testing.md
+++ b/documentation/installer/install-for-testing.md
@@ -32,14 +32,14 @@ You need to
 Set debug level for extended debugging info during installation.
 
 ```console
-sudo ansible-playbook site.yml -e "debug_level='2'" -K
+ansible-playbook site.yml -e "debug_level='2'" -K
 ```
 ## Running integration tests after installation/upgrade
 
 To only run tests (for an existing installation) use tags as follows:
 
 ```console
-sudo ansible-playbook site.yml --tags test -K
+ansible-playbook site.yml --tags test -K
 ```
 
 ## Running unit tests only
@@ -47,7 +47,7 @@ sudo ansible-playbook site.yml --tags test -K
 To only run tests (for an existing installation, can only be combined with installation_mode=upgrade) use tags as follows:
 
 ```console
-sudo ansible-playbook site.yml --tags unittest -e "installation_mode=upgrade" -K
+ansible-playbook site.yml --tags unittest -e "installation_mode=upgrade" -K
 ```
 
 ## Parameter "api_no_metadata" to prevent meta data import
@@ -55,7 +55,7 @@ sudo ansible-playbook site.yml --tags unittest -e "installation_mode=upgrade" -K
 e.g. if your hasura metadata file needs to be re-created from scratch, then use the following switch::
 
 ```console
-sudo ansible-playbook -e "api_no_metadata=yes" site.yml -K
+ansible-playbook -e "api_no_metadata=yes" site.yml -K
 ```
 
 ## Parameter "add_demo_data" to avoid creation of sample data (i.e. in production)
@@ -63,7 +63,7 @@ sudo ansible-playbook -e "api_no_metadata=yes" site.yml -K
 The following command prevents the creation of sample data in the database:
 
 ```console
-sudo ansible-playbook -e "add_demo_data=no" site.yml -K
+ansible-playbook -e "add_demo_data=no" site.yml -K
 ```
 
 note: demo/sample data can also be removed via settings menues.
@@ -73,7 +73,7 @@ note: demo/sample data can also be removed via settings menues.
 if you want to install a second ldap database "dc=example,dc=com"
 
 ```console
-sudo ansible-playbook -e "second_ldap_db=yes" site.yml -K
+ansible-playbook -e "second_ldap_db=yes" site.yml -K
 ```
 
 ### Parameter "sample_data_rate" to ramp up sample data
@@ -81,12 +81,12 @@ sudo ansible-playbook -e "second_ldap_db=yes" site.yml -K
 if you want to create sample-data changes every minute set sample_data_rate to high
 
 ```console
-sudo ansible-playbook -e "sample_data_rate=high" site.yml -K
+ansible-playbook -e "sample_data_rate=high" site.yml -K
 ```
 ### Parameter "audit_user" to add an audit user to ldap db - useful for demo installation
 
 if you want to have an extra read-only audit-user called e.g. auditor1, use the following command for installation:
 
 ```console
-sudo ansible-playbook -e "audit_user=auditor1 auditor_initial_pwd=<pwd>" site.yml -K
+ansible-playbook -e "audit_user=auditor1 auditor_initial_pwd=<pwd>" site.yml -K
 ```

--- a/documentation/installer/server-install.md
+++ b/documentation/installer/server-install.md
@@ -24,12 +24,12 @@ git clone https://github.com/CactuseSecurity/firewall-orchestrator.git
 3) Operating specific ansible adjustments
   - Ubuntu 18.04, Debian 10: install latest ansible before firewall orchestrator installation:
 
-        cd firewall-orchestrator; sudo ansible-playbook scripts/install-latest-ansible.yml -K
+        cd firewall-orchestrator; ansible-playbook scripts/install-latest-ansible.yml -K
 
 4) install (on localhost)
 
 ```console
-cd firewall-orchestrator; sudo ansible-playbook site.yml -K
+cd firewall-orchestrator; ansible-playbook site.yml -K
 ```
 Note: The installation (i.e. the connection to the target machine) is only done in root context (sudo) to secure the writing of temporary files when becoming a non-priviledged user (e.g. postgres).
 

--- a/documentation/installer/server-install.md
+++ b/documentation/installer/server-install.md
@@ -31,8 +31,6 @@ git clone https://github.com/CactuseSecurity/firewall-orchestrator.git
 ```console
 cd firewall-orchestrator; ansible-playbook site.yml -K
 ```
-Note: The installation (i.e. the connection to the target machine) is only done in root context (sudo) to secure the writing of temporary files when becoming a non-priviledged user (e.g. postgres).
-
 
 Enter sudo password when prompted "BECOME or SUDO password:"
 

--- a/documentation/installer/server-upgrade.md
+++ b/documentation/installer/server-upgrade.md
@@ -5,5 +5,5 @@ it is really simple:
 ```console
   cd firewall-orchestrator
   git pull
-  sudo ansible-playbook site.yml -K -e "installation_mode=upgrade"
+  ansible-playbook site.yml -K -e "installation_mode=upgrade"
 ```

--- a/inventory/group_vars/all.yml
+++ b/inventory/group_vars/all.yml
@@ -112,6 +112,7 @@ api_log_level: info
 ### apache: both package and dir name (needed both on ui and api hosts):
 webserver_package_name: apache2
 http_conf_dir: /etc/{{ webserver_package_name }}/sites-available/
+wsgi_package_name: libapache2-mod-wsgi
 
 ################# testing #########################
 test_dir: "{{ fworch_home }}/test"

--- a/inventory/group_vars/frontends.yml
+++ b/inventory/group_vars/frontends.yml
@@ -1,6 +1,4 @@
 
-# webserver_package_name: "apache2" 
-# http_conf_dir: /etc/{{ webserver_package_name }}/sites-available/
 ui_port: 5000
 ui_web_port: 443
 ui_app_name: "{{ product_name }}"

--- a/roles/api/tasks/api-apache-install-and-setup.yml
+++ b/roles/api/tasks/api-apache-install-and-setup.yml
@@ -9,22 +9,6 @@
 
     - set_fact: api_server_name="{{ ansible_hostname }}"          #  this only works for importer = backend
 
-    - name: enable apache modules proxy proxy_http ssl rewrite proxy_wstunnel
-      apache2_module: state=present name={{ item }}
-      loop:
-        - rewrite
-        - proxy
-        - proxy_http
-        - ssl
-        - proxy_wstunnel
-
-    - name: copy api httpd config file to api target
-      template:
-        src: "httpd.conf.j2"
-        dest: "{{ http_conf_dir }}/{{ product_name }}-api.conf"
-        owner: root
-        group: root
-
     - name: create apache certificates
       include_role:
         name: openssl-cert
@@ -46,7 +30,25 @@
         src: "/etc/apache2/ssl/server.crt"
         dest: "/etc/ssl/certs/"
         remote_src: yes
+      when: installation_mode == "new"
   
+    - name: enable apache modules proxy proxy_http ssl rewrite proxy_wstunnel
+      apache2_module: state=present name={{ item }}
+      loop:
+        - rewrite
+        - proxy
+        - proxy_http
+        - ssl
+        - proxy_wstunnel
+
+    - name: copy api httpd config file to api target
+      template:
+        src: "httpd.conf.j2"
+        dest: "{{ http_conf_dir }}/{{ product_name }}-api.conf"
+        owner: root
+        group: root
+
+
     - name: enable {{ product_name }} web site
       command: "a2ensite {{ product_name }}-api"
 

--- a/roles/api/tasks/api-apache-install-and-setup.yml
+++ b/roles/api/tasks/api-apache-install-and-setup.yml
@@ -4,7 +4,7 @@
       package: name={{ item }} state=present
       with_items:
         - "{{ webserver_package_name }}"
-        - libapache2-mod-wsgi
+        - "{{ wsgi_package_name }}"
       environment: "{{ proxy_env }}"
 
     - set_fact: api_server_name="{{ ansible_hostname }}"          #  this only works for importer = backend

--- a/roles/api/tasks/main.yml
+++ b/roles/api/tasks/main.yml
@@ -14,12 +14,6 @@
     group: "{{ fworch_group }}"
   become: yes
 
-- name: install acl
-  package:
-    name: acl
-    state: present
-  become: yes
-
 - name: install apache2
   import_tasks: api-apache-install-and-setup.yml
 

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -125,7 +125,7 @@
     when: ansible_facts['distribution']|lower == 'debian'
 
   - set_fact:
-      wsgi_package_name: {{ wsgi_package_name }}-py3
+      wsgi_package_name: "{{ wsgi_package_name }}-py3"
     when: ansible_facts['distribution']|lower == 'debian' and distribution_version|int>10
 
   - name: copy iso.conf to target

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -53,10 +53,13 @@
       upgrade: dist
     when: ansible_facts['distribution'] == "Red Hat" or ansible_facts['distribution'] == "CentOS"
 
-  - name: install packages rsync
+  - name: install packages rsync, acl (for non-root user in ansible)
     package:
-      name: "rsync"
+      name: "{{ item }}"
       state: present
+    loop:
+      - rsync
+      - acl
 
   - name: install rsyslog
     import_tasks: install_syslog.yml

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -6,6 +6,24 @@
         - "{{ ansible_version.string is version_compare('2.8', '>=') }}"
       msg: Ansible 2.8 or above is required
 
+  - name: update operating system packages .deb based
+    apt:
+      upgrade: dist
+    when: ansible_facts['distribution'] == "Ubuntu" or ansible_facts['distribution'] == "Debian"
+
+  - name: update operating system packages .rpm based (untested)
+    yum:
+      upgrade: dist
+    when: ansible_facts['distribution'] == "Red Hat" or ansible_facts['distribution'] == "CentOS"
+
+  - name: install packages rsync, acl (for non-root user in ansible)
+    package:
+      name: "{{ item }}"
+      state: present
+    loop:
+      - rsync
+      - acl
+
   - name: check for existing main config file {{ fworch_conf_file }}
     stat:
       path: "{{ fworch_conf_file }}"
@@ -42,24 +60,6 @@
   - name: include redhat install basics
     include_tasks: redhat_preps.yml
     when: installation_mode == "new" and ansible_os_family == "RedHat"
-
-  - name: update operating system packages .deb based
-    apt:
-      upgrade: dist
-    when: ansible_facts['distribution'] == "Ubuntu" or ansible_facts['distribution'] == "Debian"
-
-  - name: update operating system packages .rpm based (untested)
-    yum:
-      upgrade: dist
-    when: ansible_facts['distribution'] == "Red Hat" or ansible_facts['distribution'] == "CentOS"
-
-  - name: install packages rsync, acl (for non-root user in ansible)
-    package:
-      name: "{{ item }}"
-      state: present
-    loop:
-      - rsync
-      - acl
 
   - name: install rsyslog
     import_tasks: install_syslog.yml
@@ -121,12 +121,12 @@
       group: "{{ fworch_group }}"
       mode: '0700'
 
-  - set_fact: distribution_version="{{ ansible_facts['distribution_version'] | regex_replace('\.\d$','') }}"
-    when: ansible_facts['distribution']|lower == 'debian'
-
   - set_fact:
       wsgi_package_name: "{{ wsgi_package_name }}-py3"
-    when: ansible_facts['distribution']|lower == 'debian' and distribution_version|int>10
+    when: | 
+      (ansible_facts['distribution']|lower == 'debian' and ansible_facts['distribution_version'] is version_compare('10', '>'))
+      or 
+      (ansible_facts['distribution']|lower == 'ubuntu' and ansible_facts['distribution_version'] is version_compare('20', '>'))
 
   - name: copy iso.conf to target
     copy:

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -121,6 +121,13 @@
       group: "{{ fworch_group }}"
       mode: '0700'
 
+  - set_fact: distribution_version="{{ ansible_facts['distribution_version'] | regex_replace('\.\d$','') }}"
+    when: ansible_facts['distribution']|lower == 'debian'
+
+  - set_fact:
+      wsgi_package_name: {{ wsgi_package_name }}-py3
+    when: ansible_facts['distribution']|lower == 'debian' and distribution_version|int>10
+
   - name: copy iso.conf to target
     copy:
       src: iso.conf

--- a/roles/database/tasks/main.yml
+++ b/roles/database/tasks/main.yml
@@ -16,7 +16,6 @@
       loop:
         - postgresql-server
         - python3-psycopg2
-        - acl
       when: ansible_os_family == "RedHat"
       # todo: check if we need and if yes, how to install libpq-dev(el)
 
@@ -28,7 +27,6 @@
         - postgresql
         - python3-psycopg2
         - libpq-dev
-        - acl
       when: ansible_os_family == "Debian"
 
     - name: initdb as extra step for redhat

--- a/roles/docker/tasks/main.yml
+++ b/roles/docker/tasks/main.yml
@@ -12,10 +12,12 @@
       - curl
       - software-properties-common
 
-  - name: add apt signing key for docker
-    apt_key:
+  - name: adding docker apt signing key
+    get_url:
       url: https://download.docker.com/linux/ubuntu/gpg
-      state: present
+      dest: /etc/apt/trusted.gpg.d/docker.asc
+      force: yes
+      mode: "0644"
     environment: "{{ proxy_env }}"
 
   - name: add docker repo

--- a/roles/docker/tasks/upgrade/5.1.17.yml
+++ b/roles/docker/tasks/upgrade/5.1.17.yml
@@ -1,3 +1,0 @@
-- debug: 
-    msg:
-      - "running dummy common upgrade script"

--- a/roles/docker/tasks/upgrade/5.7.1.yml
+++ b/roles/docker/tasks/upgrade/5.7.1.yml
@@ -1,0 +1,35 @@
+- debug: 
+    msg:
+      - "running upgrade script docker 5.7.1"
+
+# - name: backup docker repo file
+#   copy: remote_src=True src=/etc/apt/sources.list.d/docker.list dest=/tmp/docker.list
+#   become: yes
+
+# - name: remove docker repo file temporarily
+#   file: path=/etc/apt/sources.list.d/docker.list state=absent
+#   become: yes
+
+# - name: remove old apt-key signing key for docker
+#   apt_key:
+#     url: https://download.docker.com/linux/ubuntu/gpg
+#     state: absent
+#     keyring: /etc/apt/trusted.gpg
+#   environment: "{{ proxy_env }}"
+#   become: yes
+
+- name: remove old apt-key signing key for docker using command as it does not work via apt_key module
+  command: apt-key del "9DC8 5822 9FC7 DD38 854A  E2D8 8D81 803C 0EBF CD88"
+  become: yes
+
+- name: adding docker apt signing key
+  get_url:
+    url: https://download.docker.com/linux/ubuntu/gpg
+    dest: /etc/apt/trusted.gpg.d/docker.asc
+    force: yes
+    mode: "0644"
+  environment: "{{ proxy_env }}"
+
+# - name: restore docker repo file
+#   copy: remote_src=True dest=/etc/apt/sources.list.d/docker.list src=/tmp/docker.list
+#   become: yes

--- a/roles/lib/tasks/main.yml
+++ b/roles/lib/tasks/main.yml
@@ -22,6 +22,7 @@
         - libgdiplus
         - libc6-dev
         - libjpeg62
+        - xfonts-75dpi
 
     - name: backup lib dir
       synchronize:

--- a/roles/middleware/tasks/main.yml
+++ b/roles/middleware/tasks/main.yml
@@ -56,6 +56,7 @@
       - libgdiplus
       - libc6-dev
       - libjpeg62
+      - xfonts-75dpi
   
   become: yes
 

--- a/roles/middleware/tasks/main.yml
+++ b/roles/middleware/tasks/main.yml
@@ -47,7 +47,6 @@
     loop:
       - python3-cryptography
       - python3-openssl
-      - acl
 
   - name: install libs needed for pdf creator nuget packet HakanL/WkHtmlToPdf-DotNet
     package:

--- a/roles/middleware/tasks/mw_apache_install_and_setup.yml
+++ b/roles/middleware/tasks/mw_apache_install_and_setup.yml
@@ -4,7 +4,7 @@
       package: name={{ item }} state=present
       with_items:
         - "{{ webserver_package_name }}"
-        - libapache2-mod-wsgi
+        - "{{ wsgi_package_name }}"
       environment: "{{ proxy_env }}"
 
     - name: enable apache modules proxy proxy_http ssl

--- a/roles/middleware/tasks/mw_apache_install_and_setup.yml
+++ b/roles/middleware/tasks/mw_apache_install_and_setup.yml
@@ -1,5 +1,12 @@
 
 - block:
+    - name: Install webserver packages
+      package: name={{ item }} state=present
+      with_items:
+        - "{{ webserver_package_name }}"
+        - "{{ wsgi_package_name }}"
+      environment: "{{ proxy_env }}"
+
     - name: copy mw httpd config file to mw target
       template:
         src: "httpd.conf"
@@ -23,12 +30,12 @@
         openssl_cert: "{{ openssl_cert_path }}/server.crt"
       when: installation_mode == "new"
 
-    - name: Install webserver packages
-      package: name={{ item }} state=present
-      with_items:
-        - "{{ webserver_package_name }}"
-        - "{{ wsgi_package_name }}"
-      environment: "{{ proxy_env }}"
+    - name: copy cert to /etc/ssl/certs/ to make it acceptable
+      copy:
+        src: "/etc/apache2/ssl/server.crt"
+        dest: "/etc/ssl/certs/"
+        remote_src: yes
+      when: installation_mode == "new"
 
     - name: enable apache modules proxy proxy_http ssl
       apache2_module: state=present name={{ item }}
@@ -38,7 +45,7 @@
         - proxy_http
         - ssl
         - proxy_wstunnel
-        
+
     - name: enable {{ product_name }} middleware apache site
       command: "a2ensite {{ product_name }}-mw"
       register: testoutput
@@ -48,7 +55,6 @@
         path: "/etc/apache2/ports.conf"
         line: "Listen {{ middleware_web_listener_port }}"
         insertafter: "<IfModule ssl_module>"
-
 
     - name: restart apache
       service:

--- a/roles/middleware/tasks/mw_apache_install_and_setup.yml
+++ b/roles/middleware/tasks/mw_apache_install_and_setup.yml
@@ -1,21 +1,5 @@
 
 - block:
-    - name: Install webserver packages
-      package: name={{ item }} state=present
-      with_items:
-        - "{{ webserver_package_name }}"
-        - "{{ wsgi_package_name }}"
-      environment: "{{ proxy_env }}"
-
-    - name: enable apache modules proxy proxy_http ssl
-      apache2_module: state=present name={{ item }}
-      loop:
-        - rewrite
-        - proxy
-        - proxy_http
-        - ssl
-        - proxy_wstunnel
-
     - name: copy mw httpd config file to mw target
       template:
         src: "httpd.conf"
@@ -39,6 +23,22 @@
         openssl_cert: "{{ openssl_cert_path }}/server.crt"
       when: installation_mode == "new"
 
+    - name: Install webserver packages
+      package: name={{ item }} state=present
+      with_items:
+        - "{{ webserver_package_name }}"
+        - "{{ wsgi_package_name }}"
+      environment: "{{ proxy_env }}"
+
+    - name: enable apache modules proxy proxy_http ssl
+      apache2_module: state=present name={{ item }}
+      loop:
+        - rewrite
+        - proxy
+        - proxy_http
+        - ssl
+        - proxy_wstunnel
+        
     - name: enable {{ product_name }} middleware apache site
       command: "a2ensite {{ product_name }}-mw"
       register: testoutput

--- a/roles/sample-data/tasks/main.yml
+++ b/roles/sample-data/tasks/main.yml
@@ -4,13 +4,12 @@
   include_tasks: setup-sample-import.yml
   when: "installation_mode=='new' or sample_role_purpose is match('test')"
 
-- name: install packages acl and postgresql-client for adding sample devices to db
+- name: install package postgresql-client for adding sample devices to db
   package:
     name: "{{ item }}"
     state: present
   loop:
     - postgresql-client
-    - acl
   become: yes
 
 - name: add localhost hostkey to known_hosts

--- a/roles/sample-data/tasks/setup-sample-import.yml
+++ b/roles/sample-data/tasks/setup-sample-import.yml
@@ -23,7 +23,7 @@
         state: present
         regex: "ansible-generated"
 
-    - name: correct sample-data users authorized_keys file acls
+    - name: correct sample-data users authorized_keys file permissions
       file:
         path: "{{ sample_config_user_home }}/.ssh/authorized_keys"
         owner: "{{ sample_config_user }}"

--- a/roles/test/handlers/main.yml
+++ b/roles/test/handlers/main.yml
@@ -1,10 +1,4 @@
 ---
-- name: install package acl
-  package:
-    name: acl
-    state: present
-  become: yes
-
 - name: delete test user and dir
   user:
     name: test

--- a/roles/ui/tasks/main.yml
+++ b/roles/ui/tasks/main.yml
@@ -34,6 +34,7 @@
         - libgdiplus
         - libc6-dev
         - libjpeg62
+        - xfonts-75dpi
       
     - name: create ui dir
       file:

--- a/roles/ui/tasks/main.yml
+++ b/roles/ui/tasks/main.yml
@@ -26,7 +26,7 @@
         path: "{{ fworch_home }}/ui"
       when: installation_mode == "upgrade"
 
-    - name: install libs needed for pdf creator nuget packet HakanL/WkHtmlToPdf-DotNet and acl
+    - name: install libs needed for pdf creator nuget packet HakanL/WkHtmlToPdf-DotNet
       package:
         name: "{{ item }}"
         state: present
@@ -34,7 +34,6 @@
         - libgdiplus
         - libc6-dev
         - libjpeg62
-        - acl
       
     - name: create ui dir
       file:

--- a/roles/ui/tasks/ui_apache_install_and_setup.yml
+++ b/roles/ui/tasks/ui_apache_install_and_setup.yml
@@ -2,9 +2,9 @@
 - block:
     - name: Install webserver packages
       package: name={{ item }} state=present
-      with_items:
+      loop:
         - "{{ webserver_package_name }}"
-        - libapache2-mod-wsgi
+        - "{{ wsgi_package_name }}"
       environment: "{{ proxy_env }}"
 
     - name: enable apache modules proxy proxy_http ssl

--- a/roles/ui/tasks/ui_apache_install_and_setup.yml
+++ b/roles/ui/tasks/ui_apache_install_and_setup.yml
@@ -7,22 +7,6 @@
         - "{{ wsgi_package_name }}"
       environment: "{{ proxy_env }}"
 
-    - name: enable apache modules proxy proxy_http ssl
-      apache2_module: state=present name={{ item }}
-      loop:
-        - rewrite
-        - proxy
-        - proxy_http
-        - ssl
-        - proxy_wstunnel
-
-    - name: copy ui httpd config file to ui target
-      template:
-        src: "httpd.conf"
-        dest: "{{ http_conf_dir }}/{{ product_name }}-ui.conf"
-        owner: root
-        group: root
-        
     - name: create apache certificates
       include_role:
         name: openssl-cert
@@ -38,6 +22,29 @@
         openssl_email_address: "fworch@cactus.de"
         openssl_cert: "{{ openssl_cert_path }}/server.crt"
       when: installation_mode == "new"
+
+    - name: copy cert to /etc/ssl/certs/ to make it acceptable
+      copy:
+        src: "/etc/apache2/ssl/server.crt"
+        dest: "/etc/ssl/certs/"
+        remote_src: yes
+      when: installation_mode == "new"
+
+    - name: enable apache modules proxy proxy_http ssl
+      apache2_module: state=present name={{ item }}
+      loop:
+        - rewrite
+        - proxy
+        - proxy_http
+        - ssl
+        - proxy_wstunnel
+
+    - name: copy ui httpd config file to ui target
+      template:
+        src: "httpd.conf"
+        dest: "{{ http_conf_dir }}/{{ product_name }}-ui.conf"
+        owner: root
+        group: root
 
     - name: enable {{ product_name }} web site
       command: "a2ensite {{ product_name }}-ui"

--- a/roles/webhook/templates/fworch-webhook-receiver.py.j2
+++ b/roles/webhook/templates/fworch-webhook-receiver.py.j2
@@ -109,14 +109,14 @@ def post_handler():
                   # might not work due to user rights:
                   # os.system("systemctl restart fworch-webhook-receiver.service")
 {% if webhook_install_mode == 'reinstall' %}
-               uninstall_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && sudo ansible-playbook site.yml -e \"installation_mode=uninstall " + prox + "\""
+               uninstall_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && ansible-playbook site.yml -e \"installation_mode=uninstall " + prox + "\""
                f.write('executing uninstall command: ' + uninstall_cmd + '\n')
                os.system(uninstall_cmd) # remove fworch
                # fresh install: 
-               build_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && sudo ansible-playbook site.yml -e \"testkeys=yes installation_mode=new " + prox + "\""
+               build_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && ansible-playbook site.yml -e \"testkeys=yes installation_mode=new " + prox + "\""
 {% else %}
                # upgrade 
-               build_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && sudo ansible-playbook site.yml -e \"testkeys=yes installation_mode=upgrade " + prox + "\""
+               build_cmd = "cd " + tmp_git_dir + "/firewall-orchestrator && ansible-playbook site.yml -e \"testkeys=yes installation_mode=upgrade " + prox + "\""
 {% endif %}
                f.write('executing build command: ' + build_cmd + '\n')
                os.system(build_cmd) # building fworch

--- a/scripts/devsrv-manual-install.sh
+++ b/scripts/devsrv-manual-install.sh
@@ -2,5 +2,5 @@ cd
 rm -rf firewall-orchestrator
 ssh-agent bash -c 'ssh-add .ssh/id_github_deploy && git clone git@github.com:CactuseSecurity/firewall-orchestrator.git'
 cd firewall-orchestrator
-sudo ansible-playbook site.yml -e "installation_mode=uninstall"
-sudo ansible-playbook site.yml -e "installation_mode=new"
+ansible-playbook site.yml -e "installation_mode=uninstall"
+ansible-playbook site.yml -e "installation_mode=new"

--- a/scripts/install_toplevel.sh
+++ b/scripts/install_toplevel.sh
@@ -22,4 +22,4 @@ ansible -m ping 127.0.0.1
 git clone https://github.com/CactuseSecurity/firewall-orchestrator.git
 
 #install firewall-orchestrator
-cd firewall-orchestrator || exit; sudo ansible-playbook site.yml -K
+cd firewall-orchestrator || exit; ansible-playbook site.yml -K

--- a/scripts/preinstall-packages.yml
+++ b/scripts/preinstall-packages.yml
@@ -42,13 +42,13 @@
 
 ########## docker for api ##################
 
-    - name: add apt signing key for docker
-      apt_key:
+    - name: adding docker apt signing key
+      get_url:
         url: https://download.docker.com/linux/ubuntu/gpg
-        state: present
-      environment:
-        https_proxy: "{{ https_proxy }}"
-      when: https_proxy is defined
+        dest: /etc/apt/trusted.gpg.d/docker.asc
+        force: yes
+        mode: "0644"
+      environment: "{{ proxy_env }}"
 
     - name: add docker repo
       lineinfile:
@@ -70,13 +70,17 @@
 
 ########## dotnet ##################
 
-    # - set_fact:
-    #     distribution_version: "{{ ansible_facts['distribution_version'] }}"
-    #   when: ansible_facts['distribution']|lower == 'ubuntu'
+    - set_fact: 
+        distribution_version: "debian/11"
+        when: ansible_facts['distribution']|lower == 'debian' and ansible_facts['distribution_version'] == '11'
 
     - set_fact: 
         distribution_version: "debian/10"
-        when: ansible_facts['distribution']|lower == 'debian'
+        when: ansible_facts['distribution']|lower == 'debian' and ansible_facts['distribution_version'] == '10'
+
+    - set_fact: 
+        distribution_version: "ubuntu/22.04"
+        when: ansible_facts['distribution']|lower == 'ubuntu' and ansible_facts['distribution_version'] == '22.04'
 
     - set_fact: 
         distribution_version: "ubuntu/20.04"

